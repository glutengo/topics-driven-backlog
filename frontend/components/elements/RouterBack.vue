<script>
export default {
    name: 'RouterBack',

    props: {
        tag: {
            type: String,
            default: 'a',
        },
    },
    
    render(createElement) {
        const $root = this.$root;

        return createElement(
            // Tag
            this.tag,

            // Attributes
            {
                attrs: Object.assign(
                    {
                        href: this.tag === 'a' ? 'javascript:' : null,
                        type: this.tag === 'button' ? 'button' : null,
                    },
                    this.$attrs
                ),
                on: {
                    click() {
                        $root.$router.back();
                    },
                },
            },

            // Children
            this.$slots.default
        );
    },
};
</script>