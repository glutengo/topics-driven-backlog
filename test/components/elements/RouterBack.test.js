import { shallow } from 'vue-test-utils';
import mockRouter from 'mock-vue-router';
import RouterBack from '@/components/elements/RouterBack';

const { $router, $route } = mockRouter(['/test', '/test2'], '/test');

describe('RouterBack.test.js', () => {
    let cmp;

    beforeEach(() => {
        cmp = shallow(RouterBack, {
            mocks: {
                $router,
                $route,
            },
        });
    });

    it('is a link by default', () => {
        expect(cmp.is('a')).toBe(true);
        expect(cmp.element).toMatchSnapshot();
    });

    it('goes back one step in the router when clicked', () => {
        cmp.vm.$router.push('/test2');
        cmp.trigger('click');
        expect(cmp.vm.$route.path).toBe('/test');
    });

    it('can also render a button', () => {
        cmp = shallow(RouterBack, {
            propsData: {
                tag: 'button',
            },
        });

        expect(cmp.is('button')).toBe(true);
    });
});
