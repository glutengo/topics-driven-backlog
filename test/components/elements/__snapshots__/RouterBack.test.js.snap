// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RouterBack.test.js is a link by default 1`] = `
<a
  href="javascript:"
/>
`;
